### PR TITLE
insert->col() and update->col() use variable argument lists

### PR DESCRIPTION
**Pull request by pavarnos**
so IDEs will not show an error if the $value parameter is used.

### CODE REVIEW

File: src/Common/Update.php [hunk 1]
```diff
@@ -119,12 +119,13 @@ public function orWhere($cond)
      *
      * @param string $col The column name.
      *
-     * @return $this
+     * @param array  $value
```
Comment (harikt) [inline src/Common/Update.php]: there is extra space, may be you could remove it also for the sake of consistency ?

File: src/Common/ValuesInterface.php [hunk 2]
```diff
@@ -24,10 +24,11 @@
      *
      * @param string $col The column name.
      *
-     * @return $this
+     * @param array  $value
```
Comment (harikt) [inline src/Common/ValuesInterface.php]: Here also.

File: src/Common/Update.php (hunk 1 above)
Comment (pavarnos) [reply src/Common/Update.php]: i looked at the code and some `@params` had a blank line between, some did not. What is your preference? I'll fix the others too...

Comment (harikt) [reply src/Common/Update.php]: I think one space is enough for now. I was also looking at the old code.

Comment (pavarnos) [reply src/Common/Update.php]: still not quite understanding: should `@param` have one blank line between each declaration, or should they be on consecutive lines with no blank line between them

Comment (pavarnos) [reply src/Common/Update.php]: eg see https://github.com/auraphp/Aura.SqlQuery/blob/2.x/src/Common/Update.php#L83 vs https://github.com/auraphp/Aura.SqlQuery/blob/2.x/src/Common/Update.php#L153

Comment (harikt) [reply src/Common/Update.php]: @pavarnos I was talking about space `@param array  $value` vs `@param array $value` . I missed to understand you were talking about extra line in b/w @param. You can have an extra line in b/w @param or we can leave that for now :) .

Comment (pavarnos) [reply src/Common/Update.php]: ah. silly me. fixed now (3 places)

File: src/Common/Update.php [hunk 3]
```diff
@@ -119,12 +119,13 @@ public function orWhere($cond)
...
-     * @return $this
+     * @param array $value
      *
+     * @return $this
      */
-    public function col($col)
+    public function col($col, ...$value)
     {
-        return call_user_func_array(array($this, 'addCol'), func_get_args());
+        return empty($value) ? $this->addCol($col) : $this->addCol($col, $value[0]);
```
Comment (harikt) [inline src/Common/Update.php]: I like this may be more performance ?

Comment (pavarnos) [reply src/Common/Update.php]: don't know. i wanted to make the function signature more explicit so it was easier for IDEs to auto-fill / auto-complete. There is another trickier one in Update.php https://github.com/auraphp/Aura.SqlQuery/blob/2.x/src/Common/Update.php#L89 which i will do if this PR gets merged. 
Got a few other PRs planned eg a fix for #100 but will start getting merge conflicts if i make too many new PRs before the current ones get merged or rejected...

File: src/Common/Insert.php [hunk 4]
```diff
@@ -160,14 +160,14 @@ public function getLastInsertIdName($col)
...
+     * @param array $value Optional: a value to bind to the placeholder.
      *
      * @return $this
      *
      */
-    public function col($col)
+    public function col($col, ...$value)
     {
-        return call_user_func_array(array($this, 'addCol'), func_get_args());
+        return empty($value) ? $this->addCol($col) : $this->addCol($col, $value[0]);
```
Comment (harikt) [inline src/Common/Insert.php]: Actually can't we just use `$this->addCol($col, ...$params);` here? Tried https://3v4l.org/sJ6S0 .

File: src/Common/Update.php (hunk 3 above)
Comment (harikt) [reply src/Common/Update.php]: Actually you can just use `$this->addCol($col, ...$params);` . I have tried https://3v4l.org/sJ6S0 . I can merge some of the PR's . The interface changes probably need more review from @pmjones .

Comment (pavarnos) [reply src/Common/Update.php]: that is much cleaner! thanks!

**Comment by harikt** [comment]
Thank you.

**Pull request merged**